### PR TITLE
lib: Folder marker is now a folder

### PR DESCRIPTION
**Pull request by AudriusButkevicius**
So we can store all sorts of stuff there.
Also moves some functions to fs package.

### CODE REVIEW

**Comment by calmh** [comment]
@st-review lgtm from the phone, someone else to agree or I'll merge it from a computer later

**Comment by st-review** [comment]
@calmh: Noted! Need another LGTM or explicit merge command.

**Comment by calmh** [comment]
@st-review merge

**Comment by st-review** [comment]
:ok_hand: Merged as ab132ff6fe056324a023d89324350427e14c09e0. Thanks, @AudriusButkevicius!

File: lib/config/testdata/v23.xml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<configuration version="22">
```
Comment (imsodin) [inline lib/config/testdata/v23.xml]: Forgot to increment number. Shouldn't a test pick this up?

Comment (AudriusButkevicius) [reply lib/config/testdata/v23.xml]: It doesn't matter, the test does the migration, we just check that things after the migration are in the right places where we expect them to be.

Comment (calmh) [reply lib/config/testdata/v23.xml]: and specifically that test is pointless busywork when the config format hasn't changed. it was good the first few iterations and we changed things like how we represent devices etc.

**Pull request closed without merging**